src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -56,13 +57,22 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner = owner;
                 _subjectBuffer = subjectBuffer;
 
+                // track context changes
+                _registration = Workspace.GetWorkspaceRegistration(_subjectBuffer.AsTextContainer());
+                _registration.WorkspaceChanged += OnWorkspaceChanged;
```

Comment:
This isn't unsubscribed from and thus a leak.

Reply:
when tagger goes away, registration also goes away, so, it won't be leak, but sure, I can be explicit and unsubscribe at the dispose.